File: solutions/euphoria/difference-of-squares/1/difference-of-squares.ex
public function squareOfSum(integer n)
    atom sum = 0
    for i = 1 to n do
        sum += i
    end for
    return power(sum, 2)
end function

public function sumOfSquares(integer n)
    atom sum = 0
    for i = 1 to n do
        sum += power(i, 2)
    end for
    return sum
end function

public function differenceOfSquares(integer n)
    return squareOfSum(n) - sumOfSquares(n)
end function

File: solutions/swift/difference-of-squares/1/Sources/DifferenceOfSquares/DifferenceOfSquares.swift
struct Squares {
    private let limit: Int
    
    init(_ limit: Int) {
        self.limit = limit
    }
    
    var squareOfSum: Int {
        let sum = (limit * (1 + limit)) / 2
        return sum * sum
    }

    var sumOfSquares: Int {
        (limit * (limit + 1) * (2 * limit + 1)) / 6
    }

    var differenceOfSquares: Int {
        squareOfSum - sumOfSquares
    }
}